app/Http/Controllers/CarController.php

```diff
@@ -76,14 +88,29 @@ public function edit(string $id)
     public function update(Request $request, string $id)
     {
         //dd($request,$id);
-        $data =[
-            'carTable' => $request->carTable ,
-            'description'=>$request->description,
-            'price'=>$request->price,
-            'published'=>isset($request->published)
-        ];
+        $data=$request->validate([
+            'carTable' => 'required|string',
+            'description'=>'required|string|max:100',
+            'price'=>'required|numeric',
+            'image'=>'nullable|image|mimes:png,jpg,jpeg,gif|max:2024'
+        ]);
+        $car=Car::findOrfail($id);
+        $car->carTable = $request->name;
+        $car->description = $request->content;
+        $car->price = $request->name;
+        $data['published']= isset($request->published);
+        if($request['image']=='true'){
+        $file_extension = $request->image->getClientOriginalExtension();
```

Comment:
not a good idea to duplicate the same code twice, was it better to make another function to handle file upload


but don't worry,, we will make a pretty good solution in the session